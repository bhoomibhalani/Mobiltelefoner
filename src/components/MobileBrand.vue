<script>
import {mdiHeart} from '@mdi/js';

export default {
  name: 'Brand',
  props: {
    brand: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      wishList: mdiHeart,
    };
  },
  computed: {
    iconColor() {
      return this.brand.active ? '#b71c1c' : 'black';
    },
  },
  methods: {
    saveMobile() {
      this.$emit('save-mobile', {...this.brand, active: !this.brand.active});
    },
    getBrandDetail() {
      this.$emit('redirect-to-brand-detail', this.brand.id);
    },
  },
};
</script>

<template>
  <v-card>
    <v-btn class="wish-list" color="white" icon @click="saveMobile">
      <v-icon :color="iconColor">{{ wishList }}</v-icon>
    </v-btn>
    <v-img
      :contain="true"
      :src="brand.imgUrl"
      height="300"
      :alt="brand.displayName"
      :title="brand.displayName"
      @click="getBrandDetail"
    />

    <v-card-text>
      <div class="font-weight-bold ml-8 mb-2" @click="getBrandDetail">
        <h3>{{ brand.displayName }}</h3>
      </div>
    </v-card-text>
  </v-card>
</template>

<style lang="scss" scoped>
@import '../sass/color.scss';
.wish-list {
  float: right;
  top: 15px;
  right: 15px;
}
.wish-list svg {
  fill: $wish-list-color;
}
.v-responsive {
  width: 100%;
}
.ma-0 {
  margin: 0;
}
</style>
