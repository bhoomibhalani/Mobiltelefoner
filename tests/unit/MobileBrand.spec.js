import MobileBrand from '@/components/MobileBrand.vue';
import {createLocalVue, shallowMount} from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

const getWrapper = () => {
  const localVue = createLocalVue();

  localVue.use(Vuex);

  const store = new Vuex.Store({
    state: {},
    modules: {
      phone: {
        namespaced: true,
        actions: {
          setActivePhone: jest.fn(),
        },
      },
    },
  });
  return shallowMount(MobileBrand, {
    localVue,
    store,
    vuetify: new Vuetify(),
    propsData: {
      brand: {displayName: 'Apple', active: true, imgUrl: 'http://www.fakeimg.com'},
      isMobileSaveLocally: false,
    },
  });
};
describe('MobileBrand.vue', () => {
  test('renders snapshots', () => {
    const wrapper = getWrapper();
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('renders snapshots when wishlist feature is on', () => {
    const wrapper = getWrapper();
    wrapper.setProps({isMobileSaveLocally: true});
    expect(wrapper.html()).toMatchSnapshot();
  });
  describe('Methods', () => {
    test('saveMobile function', () => {
      const wrapper = getWrapper();
      wrapper.vm.$emit = jest.fn();
      const saveMobile = jest.spyOn(wrapper.vm, 'saveMobile');
      wrapper.vm.saveMobile();
      expect(saveMobile).toHaveBeenCalledTimes(1);
       expect(wrapper.vm.$emit).toHaveBeenCalledTimes(1);
    });
    test('getBrandDetail function', () => {
      const wrapper = getWrapper();
      wrapper.vm.$emit = jest.fn();
      const getBrandDetail = jest.spyOn(wrapper.vm, 'getBrandDetail');
      wrapper.vm.getBrandDetail();
      expect(getBrandDetail).toHaveBeenCalledTimes(1);
      expect(wrapper.vm.$emit).toHaveBeenCalledTimes(1);
    });
  });
  describe('computed property', () => {
    test('`iconColor` when active is true', () => {
      const wrapper = getWrapper();
      expect(wrapper.vm.iconColor).toBe('#b71c1c');
    });
  });
});
