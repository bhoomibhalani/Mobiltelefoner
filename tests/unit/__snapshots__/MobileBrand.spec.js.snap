// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileBrand.vue renders snapshots 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-btn-stub color="white" tag="button" activeclass="" icon="true" type="button" class="wish-list">
    <v-icon-stub color="#b71c1c">M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z</v-icon-stub>
  </v-btn-stub>
  <v-img-stub height="300" alt="Apple" contain="true" options="[object Object]" position="center center" src="http://www.fakeimg.com" transition="fade-transition" title="Apple"></v-img-stub>
  <v-card-text-stub>
    <div class="font-weight-bold ml-8 mb-2">
      <h3>Apple</h3>
    </div>
  </v-card-text-stub>
</v-card-stub>
`;

exports[`MobileBrand.vue renders snapshots when wishlist feature is on 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-btn-stub color="white" tag="button" activeclass="" icon="true" type="button" class="wish-list">
    <v-icon-stub color="#b71c1c">M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z</v-icon-stub>
  </v-btn-stub>
  <v-img-stub height="300" alt="Apple" contain="true" options="[object Object]" position="center center" src="http://www.fakeimg.com" transition="fade-transition" title="Apple"></v-img-stub>
  <v-card-text-stub>
    <div class="font-weight-bold ml-8 mb-2">
      <h3>Apple</h3>
    </div>
  </v-card-text-stub>
</v-card-stub>
`;
